src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Repositories/BaseRepository.cs

```diff
@@ -1,63 +1,59 @@
 namespace Masa.BuildingBlocks.Ddd.Domain.Repositories;
-public abstract class BaseRepository<TEntity> : IRepository<TEntity>, IUnitOfWork
-    where TEntity : class, IAggregateRoot
+public abstract class BaseRepository<TAggregateRoot> :
+    IRepository<TAggregateRoot>, IUnitOfWork
+    where TAggregateRoot : class, IAggregateRoot
```

Comment:
Why not TEntity

Reply:
Repository should operate on the aggregate root instead of the entity. TEntity will make people understand that the operation is the entity